### PR TITLE
Setup layout for thread and comment pages

### PR DESCRIPTION
**Pull request by jybao1**
This PR is to be merged after #34.

## Changes:
- Create a ``thread-list`` component to display a list of threads . Show timestamp, primaryTag and topic of each thread.
- Created a ``thread-detail`` component to display detaled information(comments, statistics) of a single thread. Hence, with routing enabled in #34, user can click on a single thread in thread-list and navigate into each thread. 

## Screenshots:
- Show a list of threads on clicking search.
screencapture-localhost-4200-threads-2020-07-25-01_42_12

- When clicking on each thread, jump to thread detail page with a list of comments and tag statistics.
screencapture-localhost-4200-thread-30a57bf9-1d3c-4be4-b7c9-3fee77a6fce2-2020-07-25-01_32_37

### CODE REVIEW

File: ui/src/app/panel/thread-list/thread-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  selectedId: number;
+
+  constructor(private dataService: DataService,
+              private route: ActivatedRoute) {
+  }
+
+  ngOnInit(): void {
+    this.threadList$ = this.route.paramMap.pipe(
+      switchMap(params => {
+        this.selectedId = +params.get('id');
```
Comment (XuZehua-Teapot) [inline ui/src/app/panel/thread-list/thread-list.component.ts:22]: Prefer explicit notations like `Number(params.get('id'))`.

File: ui/src/app/panel/thread-list/thread-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import {switchMap} from 'rxjs/operators';
+import {DataService} from '../../service/data.service';
+
+@Component({
+  templateUrl: './thread-list.component.html',
+  styleUrls: ['./thread-list.component.sass']
+})
+export class ThreadListComponent implements OnInit {
+  threadList$: Observable<any>;
+  selectedId: number;
```
[the remark this answers is not in the document]
Comment (XuZehua-Teapot) [reply ui/src/app/panel/thread-list/thread-list.component.ts:13]: Also, this attribute seems unused?

File: ui/src/app/panel/thread-list/thread-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  constructor(private dataService: DataService,
+              private route: ActivatedRoute) {
+  }
+
+  ngOnInit(): void {
+    this.threadList$ = this.route.paramMap.pipe(
+      switchMap(params => {
+        this.selectedId = +params.get('id');
+        return this.dataService.threadList;
```
Comment (XuZehua-Teapot) [inline ui/src/app/panel/thread-list/thread-list.component.ts:23]: This looks like having the same effect as `this.threadList$ = this.dataService.threadList`.

File: ui/src/app/panel/thread-list/thread-list.component.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+              <br>
+            </div>
+          </div>
+        </div>
+        <div class="level-right">
+          <div class="level-item has-text-centered">
+            <div>
+              <!-- placeholder for comment count -->
+              <p class="heading">Comments</p>
+              <p class="title" style="font-size: 1.25em; color:#ADD8E6">123</p>
```
Comment (XuZehua-Teapot) [inline ui/src/app/panel/thread-list/thread-list.component.html:40]: Avoid using inline styles if you can put them into the CSS file.

File: ui/src/app/panel/thread-list/thread-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      })
+    );
+  }
+
+  /**
+   * Convert proto timestamp to readable string
+   */
+  private convertTimestamp(timestamp: any): string {
```
[the remark this answers is not in the document]
Comment (jybao1) [reply ui/src/app/panel/thread-list/thread-list.component.ts:31]: Fixed this in next PR #36

**Comment by jybao1** [comment]
Changes proposed in this PR has a large overlap with another PR #36. Problems in this PR are also addressed in #36.
I am going to close this PR.

**Pull request closed without merging**